service.go

```diff
@@ -21,7 +21,7 @@ import (
 var logSetupOnce sync.Once
 
 // SetupLogging sets up the default metrics logging.
-func SetupLogging(name, version string) error {
+func SetupLogging(name, version, env string) error {
```

Comment:
We are getting a lot of complaints about making breaking changes. In order to break it and be future proof for another request to add fields to our logging, we should be generic enough to accept an arbitrary number of fields by using a map. What do you think? If it is nil we should just disregard the fields.

Reply:
That's exactly the thought I had this morning... this would work

```
func SetupLogging(name, version, fields map[string]interface{}) error {
...
```

However making `fields` variadic would mean this would not be a breaking change

```
func SetupLogging(name, version, fields ...map[string]interface{}) error {
...
```

Reply:
This is what I've done, it's now a smaller change and not breaking 😄

Reply:
making fields variadic means that it can accept multiple maps. This seems not to be a very clear API though. What would a second map be?

Reply:
A single map keeps things simple and behaves as expected however that would be a breaking change, requiring existing users to provide `nil` if they don't want any extra fields.

Alternatively, if it was variadic I would expect one or zero maps to be provided 99.999% of the time (thus avoiding a breaking change). If two or more maps were provided, all the keys from all the maps would be used (with later keys overwriting earlier ones), however it could be a little confusing when reading the docs.

Let me know which you prefer and I'll change accordingly.

Reply:
Hard choice... I prefer a clean API, even it it breaks the API... Let's wait for @drakos74 and @tpaschalis for their view...

Reply:
would it maybe make sense to use another method ? 
```
SetupLoggingWithFields
```
So that we dont break backwards compatibility and also enhance the functionality.
If we dont want another method, i would prefer the breaking API , instead of creating a variadic argument that would only accept one element. It seems to me a bit counterintuitive.

Reply:
@drakos74 when patron setup is called i also calls again the log setup function which complicates things...

Reply:
`SetupLoggingWithFields` is a great idea, done. 

> when patron setup is called i also calls again the log setup function which complicates things...

I added `WithFields()` to the patron builder so it's not a problem (unless I've misunderstood)